parts/kubernetesagentresourcesvmas.t

```diff
@@ -15,22 +15,31 @@
       "name": "[concat(variables('{{.Name}}VMNamePrefix'), 'nic-', copyIndex(variables('{{.Name}}Offset')))]",
       "properties": {
 {{if .IsCustomVNET}}
-	    "networkSecurityGroup": {
-		    "id": "[variables('nsgID')]"
-	    },
+        "networkSecurityGroup": {
+          "id": "[variables('nsgID')]"
+        },
 {{end}}
         "ipConfigurations": [
+          {{range $seq := loop 1 .IPAddressCount}}
           {
-            "name": "ipconfig1",
+            "name": "ipconfig{{$seq}}",
             "properties": {
+              {{if eq $seq 1}}
+              "primary": true,
+              {{end}}
               "privateIPAllocationMethod": "Dynamic",
               "subnet": {
-                "id": "[variables('{{.Name}}VnetSubnetID')]"
-             }
+                "id": "[variables('{{$.Name}}VnetSubnetID')]"
+              }
             }
           }
-        ],
+          {{if lt $seq $.IPAddressCount}},{{end}}
+          {{end}}
+        ]
+{{if not IsVNETIntegrated}}
```

Comment:
CNI enabled is true when Calico is set, but that's not the case here. "AzureNetworkPolicy" is not an accurate descriptor either. That having been said, `$.NetworkPlugin == "Azure"` might be a more valid way of specifying this condition.